openvalidators/reward/diversity.py

```diff
@@ -37,65 +38,63 @@ def mean_pooling( model_output, attention_mask ):
             and dividing it by the sum of input_mask_expanded after clamping its values to a minimum of 1e-9.
     """
     token_embeddings = model_output[0]
-    input_mask_expanded = (
-        attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
-    )
+    input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
     return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(
         input_mask_expanded.sum(1), min=1e-9
     )
     
 class DiversityRewardModel( BaseRewardModel ):
-    relevance_model_path = "bert-base-uncased"
+    
+    diversity_model_path = "sentence-transformers/all-mpnet-base-v2"
+    
     @property
     def name(self) -> str: return RewardModelType.diversity.value
 
     def __init__( self, device: str ):
         super().__init__()
         self.device = device
-        self.tokenizer = AutoTokenizer.from_pretrained( DiversityRewardModel.relevance_model_path )
-        self.model = AutoModel.from_pretrained( DiversityRewardModel.relevance_model_path ).to(self.device)
-
-    def get_embedding( self, message: str ) -> "torch.FloatTensor":
+        self.tokenizer = AutoTokenizer.from_pretrained( DiversityRewardModel.diversity_model_path )
```

Comment:
What is the sequence length of the model? will there be an issue if we exceed it?

Reply:
Nice catch! max_len 512, let me truncate it.